payu/models/cesm_cmeps.py

```diff
@@ -190,6 +184,72 @@ def setup(self):
                 # TODO: copied this from other models. Surely we want to exit here or something
                 print('payu: error: Unable to find mod_def.ww3 file in input directory')
 
+    def _setup_checks(self):
+        # check pelayout fits within requested cpucount
+        cpucount = int(
+            self.expt.config.get('ncpus', multiprocessing.cpu_count())
+        )
+        all_realms = self.realms
+        for realm in all_realms:
+            ntasks = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_ntasks"))
+            nthreads = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_nthreads"))
+            rootpe = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_rootpe"))
+            pestride = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_pestride"))
+
+            if nthreads > 1:
+                npes = nthreads*ntasks*pestride
```

Comment:
Why isnt npes now the last index of the pelist? This is confusing to me.

Reply:
pelist starts counting at 0 , npes starts counting at 1

ie for one task, pe is 0

Reply:
Ah I see. Thanks!